crates/sncast/src/state/state_file.rs

```diff
@@ -1,13 +1,82 @@
 #![allow(dead_code)]
 use crate::helpers::constants::STATE_FILE_VERSION;
+use crate::response::errors::StarknetCommandError;
 use crate::response::structs::{DeclareResponse, DeployResponse, InvokeResponse};
 use crate::state::hashing::generate_id;
+use crate::WaitForTransactionError;
 use anyhow::{anyhow, Context, Result};
+use cairo_felt::Felt252;
 use camino::Utf8PathBuf;
+use conversions::felt252::SerializeAsFelt252Vec;
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use std::fs;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+struct InnerStateManager {
+    state_file: Utf8PathBuf,
+    executed_transactions_prev_run: ScriptTransactionEntries,
+    executed_transactions_current_run: ScriptTransactionEntries,
+}
+
+#[derive(Default)]
+pub struct StateManager {
+    inner: Option<InnerStateManager>,
+}
+
+impl StateManager {
+    pub fn from(state_file_path: Option<Utf8PathBuf>) -> Result<Self> {
+        let res = if let Some(state_file_path) = state_file_path {
+            let executed_transactions = load_or_create_state_file(&state_file_path)?
+                .transactions
+                .unwrap_or_default();
+
+            Self {
+                inner: Some(InnerStateManager {
+                    state_file: state_file_path,
+                    executed_transactions_prev_run: executed_transactions,
+                    executed_transactions_current_run: ScriptTransactionEntries::default(),
+                }),
+            }
+        } else {
+            Self::default()
+        };
+
+        Ok(res)
+    }
+
+    #[must_use]
+    pub fn maybe_get_success_output(&self, tx_id: &str) -> Option<ScriptTransactionOutput> {
```

Comment:
I would remove the prefix `maybe` since we have a similar method named this way in `ScriptTransactionEntries`. Alternatively we can rename it in both places to something like `get_output_if_success`.